### DISCUSSION TITLE
How to reset the GroupChat after one inference?

### DISCUSSION THREAD

**Discussion by paulilioaica** [microsoft/autogen#529] [Q&A]
Hello everyone.

 **I am trying to integrate AutoGen into a inference pipeline with GroupChat.**
My main workflow is this:

1. Feed data point into AutoGen agents
2. Let them thinker back and forth
3. Get the final inference.

Finally, i want to run 1,2,3 on another data point.
Whenever I run this in a loop, for some reason, the inputs merge in some way and I reach the token limit, and I get messages like the gpt4 is only 8k but I tried to feed 129k into it.

This is somewhat related to reseting the groupchat since 
> tried to feed 129k into it.

increases with every call with precisely number of tokens of current data point.

My code is this 

```
    def reset(self):
        self.groupchat.reset()
def setup(self):
        self.groupchat = autogen.GroupChat(agents=[self.dependency_analyst, self.dependency_expert, self.dependency_guru], messages=[], max_round=5)
        self.manager = autogen.GroupChatManager(groupchat=self.groupchat, llm_config=self.llm_config)
def extract_dependencies(self, code):
        self.user_proxy.initiate_chat(self.manager, message=code)
        json_response = extract_json(self.json_formatter.generate_oai_reply(messages=[self.dependency_analyst.last_message()])[1])
        self.reset()
        return json_response
```

In my main code, I call `reset()` after each `extract_dependency()` of failure in a try catch

TL;DR
How can I completley reset the enviroment between inferences? groupchat.reset() doesn't do it.

**Comment by ramonesmith777** [comment]
https://www.youtube.com/watch?v=KZPrvbNosKg

https://www.amazon.com/dp/B0CMK6GGQ7?ref_=pe_3052080_397514860

https://www.amazon.com/dp/B0CLK2TLFJ?ref_=pe_3052080_397514860

https://www.amazon.com/dp/B0CGKV5RQC?ref_=pe_3052080_397514860

**Comment by ramonesmith777** [comment]
I hope you are all well. I'd like to share an additional job search resource with you all. It's called Pathrise. Below are two links that explain how it works along with the link for enrollment. Reach out if you have any questions. Continued good luck with your job search.

https://www.pathrise.com/apply?utm_source=ramone&utm_medium=affiliate&utm_campaign=apply_linkedin

https://www.youtube.com/watch?v=j6wd4431GIE

https://www.youtube.com/watch?v=rT0AekN6nag

**Comment by tamish-max** [comment]
`groupchat.reset()` is usually **not enough** for this pattern.

It clears the group chat message list, but the participating agents can still retain conversation state / per-peer history. That is why token usage keeps growing across iterations.

For a batched inference pipeline, the safest pattern is:
- create **fresh agent instances**
- create a **fresh `GroupChat`**
- create a **fresh `GroupChatManager`**
- run one sample
- discard the whole conversation object graph

If you want to reuse objects, you need to reset **every participating agent** as well (`user_proxy`, assistants, manager/history helpers depending on your version). In practice, though, one-run-per-fresh-conversation is much less error-prone and makes retries easier too.

So I would treat each inference as an isolated session rather than trying to partially wipe shared state after the fact.
